service/src/main/java/bio/terra/workspace/service/job/JobBuilder.java

```diff
@@ -116,29 +126,80 @@ public JobBuilder addParameter(String keyName, @Nullable Object val) {
   /**
    * Submit a job to stairway and return the jobID immediately.
    *
+   * @param changeType Specify the type of {@link ActivityLogChangedType} to be logged.
+   * @param resultClass type for the response of the flight. When null, nothing is returned from the
+   *     flight.
    * @return jobID of submitted flight
    */
-  public String submit() {
+  public <T> String submit(
+      @Nullable ActivityLogChangedType changeType, @Nullable Class<T> resultClass) {
```

Comment:
It looks like you're using `changeType = null` to mean "We don't need activity log".

Instead of null, can you use optional?

If changeType is null/Optional.empty(), don't call dao.setChangedDate